src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -643,4 +636,52 @@ private Query getConstantScoreQuery(
     Query filterQuery = getQuery(constantScoreQueryGrpc.getFilter(), indexState, docLookup);
     return new ConstantScoreQuery(filterQuery);
   }
+
+  private SpanQuery getSpanQuery(
+      com.yelp.nrtsearch.server.grpc.SpanQuery protoSpanQuery, IndexState state) {
+    List<SpanQuery> clauses = new ArrayList<>();
+
+    if (protoSpanQuery.hasSpanNearQuery()) {
+      com.yelp.nrtsearch.server.grpc.SpanNearQuery protoSpanNearQuery =
+          protoSpanQuery.getSpanNearQuery();
+      for (com.yelp.nrtsearch.server.grpc.SpanQuery protoClause :
+          protoSpanNearQuery.getClausesList()) {
+        SpanQuery luceneClause = convertToLuceneSpanQuery(protoClause, state);
```

Comment:
Is there a reason this can't just recursively call `getSpanQuery`?

Reply:
Yeah, the above method looks identical to this now. Removed it.